azurerm/resource_arm_virtual_machine_unmanaged_disks_test.go

```diff
@@ -102,6 +102,28 @@ func TestAccAzureRMVirtualMachine_basicLinuxMachine_disappears(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMVirtualMachine_basicLinuxMachineUseExistingOsDiskImage(t *testing.T) {
+	var vm, mirrorVm compute.VirtualMachine
+	ri := acctest.RandInt()
+	config := testAccAzureRMVirtualMachine_basicLinuxMachineUseExistingOsDiskImage(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineExists("azurerm_virtual_machine.test", &vm),
+					testCheckAzureRMVirtualMachineExists("azurerm_virtual_machine.mirror", &mirrorVm),
+					testCheckAzureRMVirtualMachineVHDExistence("myosdisk1.vhd", true),
+					testCheckAzureRMVirtualMachineVHDExistence("mirrorosdisk.vhd", true),
```

Comment:
we can also check the `image_uri` field is set correctly in the state here by using:

```
resource.TestCheckResourceAttrSet("azurerm_virtual_machine.mirror", "storage_os_disk.0.image_uri")
```

Reply:
I mentioned this same thing to him last night and I believe he has a commit with the `image_uri` check in it ready to push.